dart.dart

```diff
@@ -0,0 +1,44 @@
+import 'src/myFile.dart';
+
+abstract class Vehicle {
+    factory Vehicle(String type) {
+        if (type == 'skateboard') return Skateboard();
+        throw 'Can\'t create $type.';
+    }
+    num get odometer;
+}
+
+class Skateboard implements Vehicle {
+    num _odometer = 0;
+    num get odometer => _odometer;
+
+    Skateboard();
+
+    void addMiles(num increment) {
+        _odometer += increment;
+    }
+
+    @override
+    String toString() => 'Skateboard with mileage $odometer';
+}
+
+class Car implements Vehicle {
+    final int modelYear;
+    final String make;
+    final String model;
+    num _odometer = 0;
+    num get odometer => _odometer;
+
+    Car(this.modelYear, this.make, this.model);
+
+    void addMiles(num increment) {
+        _odometer += increment;
+    }
+
+    @override
+    String toString() => 'Car: $modelYear $make $model with mileage $odometer';
+}
+
+void main() {
+    var car = Car(2016, 'Honda', 'Civic');
+    print(car);
```

Comment:
Is this ok? Isn't it missing the final `}`?
```suggestion
void main() {
    var car = Car(2016, 'Honda', 'Civic');
    print(car);
}
```

Reply:
Yea crap, sorry was mostly trying to pull out examples of what it looks like vs worrying about valid code